### PR TITLE
Animation refactor

### PR DESCRIPTION
**Pull request by g-plane**

### CODE REVIEW

File: src/animation.ts [hunk 1]
```diff
@@ -86,6 +99,40 @@ export class CompositeAnimation implements IAnimation {
...
+	set paused(value) {
+		if (value) {
+			this.clock.stop();
+		} else {
+			this.clock.start();
+		}
+	}
+
+	runAnimationLoop(player: PlayerObject): void {
+		if (this.handles.size == 0) {
```
Comment (g-plane) [inline src/animation.ts]:
```suggestion
		if (this.handles.size === 0) {
```

**Comment by Hacksore** [comment]
@yushijinhun I think my only feedback would be this seems like a candidate for a major version as it's not backward compatible. `2.0.0`?

**Comment by yushijinhun** [comment]
@Hacksore You are right.

So why don't we skip v1.2 and jump to v2.0? After all, switching from javascript to typescript is definitely a big change.

**Comment by yushijinhun** [comment]
> So why don't we skip v1.2 and jump to v2.0? After all, switching from javascript to typescript is definitely a big change.

Sorry, I forgot that it is in v1.1 that we switched to typescript...not v1.2.

**Comment by yushijinhun** [comment]
So, there are three options:
1. ship this feature in v1.2 (but it makes breaking change)
2. skip v1.2 and jump to v2.0
3. delay this feature to v2.0, release v1.2 first

Personally I prefer the third one. How do you think about this?

**Comment by Hacksore** [comment]
> So, there are three options:
> 
> 1. ship this feature in v1.2 (but it makes breaking change)
> 2. skip v1.2 and jump to v2.0
> 3. delay this feature to v2.0, release v1.2 first
> 
> Personally I prefer the third one. How do you think about this?

I also agree with number three.

**Pull request merged**